cbmc/proofs/IotMqtt_SubscribeAsync/Makefile

```diff
@@ -0,0 +1,68 @@
+ENTRY=IotMqtt_SubscribeAsync
+
+# We abstract all unreachable functions to improve coverage metrics
+ABSTRACTIONS += --remove-function-body _checkRetryLimit
+ABSTRACTIONS += --remove-function-body _IotMqtt_PublishSetDup
+ABSTRACTIONS += --remove-function-body _IotMqtt_SerializeUnsubscribeCommon
+ABSTRACTIONS += --remove-function-body _IotMqtt_SerializeUnsubscribe
+ABSTRACTIONS += --remove-function-body _mqttOperation_match
+ABSTRACTIONS += --remove-function-body _mqttSubscription_setUnsubscribe
+ABSTRACTIONS += --remove-function-body _packetMatch
+ABSTRACTIONS += --remove-function-body _scheduleNextRetry
+ABSTRACTIONS += --remove-function-body _topicFilterMatch
+ABSTRACTIONS += --remove-function-body IotNetworkInterfaceClose
+ABSTRACTIONS += --remove-function-body IotNetworkInterfaceReceive
+
+OBJS += $(ENTRY)_harness.goto
+OBJS += $(MQTT)/cbmc/proofs/mqtt_state.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_api.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_helper.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_operation.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_serialize.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_subscription.goto
+OBJS += $(MQTT)/libraries/standard/mqtt/src/iot_mqtt_validate.goto
+
+# One more than actual number of subscriptions in a subscription list
+SUBSCRIPTION_COUNT_MAX=2
+DEF += -DSUBSCRIPTION_COUNT_MAX=$(SUBSCRIPTION_COUNT_MAX)
+
+# One more than actual number of operations in an operation list
+OPERATION_COUNT_MAX=2
+DEF += -DOPERATION_COUNT_MAX=$(OPERATION_COUNT_MAX)
+
+# One more than actual length of topics
+TOPIC_LENGTH_MAX=3
+DEF += -DTOPIC_LENGTH_MAX=$(TOPIC_LENGTH_MAX)
+
+# Should be 2*SUBSCRIPTION_COUNT_MAX-1
+SUBSCRIPTION_LIST_MAX=3
+DEF += -DSUBSCRIPTION_LIST_MAX=$(SUBSCRIPTION_LIST_MAX)
+
+DEF += -DCBMC_STUB_REMOVEALLMATCHES=1
```

Comment:
Comment that this is required to make use of the stub above?  Do we have to do something similar for RemoveSubscription...?  If that function is a static, the proof may not be using the stub you think it is.

Reply:
Done! We don't need it for `_IotMqtt_RemoveSubscriptionByTopicFilter`, since it's not static.